Drop support for WC < 6.8 (L-2), remove few workarounds

tomalec opened the pull request:
:warning:  This should be merged after WC 7.0.0 is released or at least after GLA with min WC bumped.

### Changes proposed in this Pull Request:

This PR drops support for WooCommerce < 6.8 (soon to be L-2).

And removes a few workarounds, to keep our code shorter and cleaner.

- https://github.com/woocommerce/google-listings-and-ads/pull/1441
- https://github.com/woocommerce/google-listings-and-ads/pull/1569


### Screenshots:

<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

Follow the test instructions form both PRs
1. Check out this branch.
2. Install the latest version of WC
3. Load any admin page to see no error there.
4. Go to wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&guide=campaign-creation-success
5. Click on "Got it"
6. No errors should be displayed.
7. Redo 3-6 for WC 6.8

### Additional details:

- Probably, we could clean up much more, like update DEWPed dependencies etc. I just picked low-hanging fruits

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Remove - Support for WC < 6.8.

tomalec committed:
Drop support for WC < 6.8 (L-2).

tomalec committed:
Remove workaround for WC < 6.5 `is_admin_page`

introduced in https://github.com/woocommerce/google-listings-and-ads/pull/1441.

We now support only WC 6.8+, and the `PageController::is_admin_page`is always there.
Suppress false positives in compatibility checker.

tomalec committed:
Remove workaround for WC< 6.6 CES compatibility

introduced in https://github.com/woocommerce/google-listings-and-ads/pull/1569.

As of WC 6.6+ there is always only named export.

eason9487 reviewed (approved):
Thanks for the cleanup. LGTM. There are two places that mention the minimum support of WC version. We could update them as well:
- [readme.txt](https://github.com/woocommerce/google-listings-and-ads/blob/996691b8efeb283f305c0f859de7a783393bffae/readme.txt#L57)
- [README.md](https://github.com/woocommerce/google-listings-and-ads/blob/996691b8efeb283f305c0f859de7a783393bffae/README.md?plain=1#L33)

📝 For the record, listing other cleanable items could be resolved along with this PR or in a later PR:
- The compatibility of `history` v4 in the [`useNavigateAwayPromptEffect` hook](https://github.com/woocommerce/google-listings-and-ads/blob/996691b8efeb283f305c0f859de7a783393bffae/js/src/hooks/useNavigateAwayPromptEffect.js).
- A [workaround](https://github.com/woocommerce/google-listings-and-ads/blob/996691b8efeb283f305c0f859de7a783393bffae/js/src/dashboard/all-programs-table-card/index.js#L124-L127) in rendering the rows of `AllProgramsTableCard` could be changed to use `rowKey` prop.

tomalec committed:
Update minimum WC version in readmes

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/1705#pullrequestreview-1125064606

tomalec committed:
Merge branch 'develop' into remove/wc-6.7

tomalec commented:
Thanks for the review, @eason9487.
I updated the readmes and created an issue to clean the remaining bits https://github.com/woocommerce/google-listings-and-ads/issues/1715
